test/integration/crud/misc_cursors.test.js

```diff
@@ -31,50 +31,29 @@ describe('Cursor', function () {
     await client.close();
   });
 
-  it('cursorShouldBeAbleToResetOnToArrayRunningQueryAgain', {
-    // Add a tag that our runner can trigger on
-    // in this case we are setting that node needs to be higher than 0.10.X to run
-    metadata: {
-      requires: { topology: ['single', 'replicaset', 'sharded'] }
-    },
+  it('should not fail on toArray and forEach calls after cursor is closed', async function () {
+    const db = client.db();
 
-    test: function (done) {
-      const configuration = this.configuration;
-      client.connect((err, client) => {
-        expect(err).to.not.exist;
-        this.defer(() => client.close());
-
-        const db = client.db(configuration.db);
-        db.createCollection('test_to_a', (err, collection) => {
-          expect(err).to.not.exist;
-
-          collection.insert({ a: 1 }, configuration.writeConcernMax(), err => {
-            expect(err).to.not.exist;
+    const collection = await db.collection('test_to_a');
+    await collection.insertMany([{ a: 1 }]);
+    const cursor = collection.find({});
 
-            const cursor = collection.find({});
-            this.defer(() => cursor.close());
+    const firstToArray = await cursor.toArray().catch(error => error);
+    expect(firstToArray).to.be.an('array');
 
-            cursor.toArray(err => {
-              expect(err).to.not.exist;
+    expect(cursor.closed).to.be.true;
 
-              // Should fail if called again (cursor should be closed)
-              cursor.toArray(err => {
-                expect(err).to.not.exist;
+    // should not fail instead return empty array
+    const secondToArray = await cursor.toArray().catch(error => error);
+    expect(secondToArray).to.be.an('array');
```

Comment:
```suggestion
    expect(secondToArray).to.be.an('array');
    expect(secondToArray).to.have.lengthOf(0);
```

Reply:
we should also assert that the array is actually empty

Reply:
good point, added